ARCHITECTURE.md

```diff
@@ -0,0 +1,76 @@
+# Architectural Haiku
+
```

Comment:
Maybe add a link to what a Haiku is? In case people think you've just written a poem.